q2_vsearch/_join_pairs.py

```diff
@@ -0,0 +1,135 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os.path
+import yaml
+
+import pandas as pd
+from q2_types.per_sample_sequences import (
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt,
+    FastqManifestFormat, YamlFormat)
+
+from ._cluster_features import run_command
+
+
+def join_pairs(demultiplexed_seqs: SingleLanePerSamplePairedEndFastqDirFmt,
+               truncqual: int=None,
+               minlen: int=1,
+               maxns: int=None,
+               allowmergestagger: bool=False,
+               minovlen: int=10,
+               maxdiffs: int=10,
+               minmergelen: int=None,
+               maxmergelen: int=None,
+               maxee: float=None,
+               qmin: int=0,
+               qminout: int=0,
+               qmax: int=41,
+               qmaxout: int=41,
+               ) -> SingleLanePerSampleSingleEndFastqDirFmt:
+    _, result = _join_pairs_w_command_output(
+        demultiplexed_seqs, truncqual, minlen, maxns, allowmergestagger,
+        minovlen, maxdiffs, minmergelen, maxmergelen, maxee, qmin, qminout,
+        qmax, qmaxout)
+    return result
+
+
+def _join_pairs_w_command_output(
+        demultiplexed_seqs: SingleLanePerSamplePairedEndFastqDirFmt,
+        truncqual: int=None,
+        minlen: int=1,
+        maxns: int=None,
+        allowmergestagger: bool=False,
+        minovlen: int=10,
+        maxdiffs: int=10,
+        minmergelen: int=None,
+        maxmergelen: int=None,
+        maxee: float=None,
+        qmin: int=0,
+        qminout: int=0,
+        qmax: int=41,
+        qmaxout: int=41,
+        ) -> SingleLanePerSampleSingleEndFastqDirFmt:
```

Comment:
This return annotation is incorrect --- the helper returns a tuple of `(list, SingleLanePerSampleSingleEndFastqDirFmt)`. I think at this point I would rather see the annotations disappear from this function.